include/swift/AST/DiagnosticsSema.def

```diff
@@ -4098,6 +4098,14 @@ WARNING(class_super_not_usable_from_inline_warn,none,
         "should be '@usableFromInline' or public",
         (bool))
 
+WARNING(meta_metatype,none,
+        "meta-metatype is probably not what you want", ())
```

Comment:
IMO the wording of this diagnostic doesn’t quite fit the style of the existing diagnostics we emit in unexpected situations. I think it should be something along the lines of “type of {thing} is a meta-metatype, which may be unexpected”.

Reply:
Sure. That kind of feedback is what I initially requested and wanted when I opened this pull request. Thanks.